lib/web_ui/lib/src/engine/view_embedder/application_dom/custom_element_application_dom.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../../dom.dart';
+import 'application_dom.dart';
+
+class CustomElementApplicationDom extends ApplicationDom {
+
+  CustomElementApplicationDom(this._hostElement) {
+    // Clear children...
+    while(_hostElement.firstChild != null) {
```

Comment:
Should we instead assert that the host element is empty?

Reply:
I think I'm going to do both, actually :P

Reply:
TL;DR: Asserting _hostElement.children.isEmpty here breaks hot restart :)